torchaudio/datasets/musdb_hq.py

```diff
@@ -0,0 +1,103 @@
+import os
+from pathlib import Path
+from typing import Tuple, Union, List, Optional
+
+import torch
+import torchaudio
+from torch.utils.data import Dataset
+
+_URL = "https://zenodo.org/record/3338373/files/musdb18hq.zip"
+_EXT = ".wav"
+_SAMPLE_RATE = 44100
+
+
+class MUSDB_HQ(Dataset):
+    """Create *MUSDB_HQ Dataset* [:footcite:`MUSDB18HQ`] Dataset
+
+    Args:
+        root (str or Path): Root directory where the dataset's top level directory is found
+        subset (str): Subset of the dataset to use. Options: [``"train"``, ``"test"``].
+        sources (List[str] or None, optional): Sources extract data from.
+            Options: [``bass``, ``drums``, ``other``, ``mixture``, ``vocals``].
+            If ``None``, dataset consists of tracks except mixture.
+            (default: ``"[``bass``, ``drums``, ``other``, ``vocals``]"``)
+        split (str, optional): Whether to split training set into train and validation set.
+            If None, no splitting occurs. If ``train`` or ``validation``, returns respective set.
+            (default: ``None``)
+        validation (List[str] or None, optional): Tracks which are included in validation set.
+            If ``None``, dataset consists of no tracks in validation set.
```

Comment:
I think this is clearer:

```suggestion
        validation (List[str] or None, optional): Tracks included in the validation set, to be used for splitting.
            List must be provided if `split` is provided, or `None` if `split` is not provided.
```